src/Explorer/QueryCopilot/V2/Footer/Footer.tsx

```diff
@@ -29,18 +37,55 @@ export const Footer: React.FC = (): JSX.Element => {
   const handleEnterKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
     if (event.key === "Enter" && !event.shiftKey) {
       event.preventDefault();
-      handleSendMessage();
+      startSentMessageProcess();
     }
   };
 
-  const handleSendMessage = () => {
+  const handleSentQueryRequest = async (): Promise<void> => {
     if (userPrompt.trim() !== "") {
-      setChatMessages([...chatMessages, userPrompt]);
-      setUserPrompt("");
       setIsGeneratingQuery(true);
+      useTabs.getState().setIsTabExecuting(true);
+      useTabs.getState().setIsQueryErrorThrown(false);
+      setChatMessages([...chatMessages, { source: 0, message: userPrompt }]);
+      try {
+        const response = await SendQueryRequest({ userPrompt, explorer });
+
+        const generateSQLQueryResponse: GenerateSQLQueryResponse = await response?.json();
+        if (response.status === 404) {
+          setShouldAllocateContainer(true);
+        }
+        if (response.ok) {
+          if (generateSQLQueryResponse?.sql) {
+            let query = `Here is a query which will help you with provided prompt.\r\n **Prompt:** ${userPrompt}`;
+            query += `\r\n${generateSQLQueryResponse.sql}`;
+            setChatMessages([
+              ...chatMessages,
+              { source: 0, message: userPrompt },
+              { source: 1, message: query, explanation: generateSQLQueryResponse.explanation },
+            ]);
+            setGeneratedQuery(generateSQLQueryResponse.sql);
+            setGeneratedQueryComments(generateSQLQueryResponse.explanation);
+          }
+        } else {
+          handleError(JSON.stringify(generateSQLQueryResponse), "copilotInternalServerError");
+          useTabs.getState().setIsQueryErrorThrown(true);
+        }
+      } catch (error) {
+        handleError(error, "executeNaturalLanguageQuery");
+        useTabs.getState().setIsQueryErrorThrown(true);
+        throw error;
+      } finally {
+        setUserPrompt("");
+        setIsGeneratingQuery(false);
+        useTabs.getState().setIsTabExecuting(false);
+      }
     }
   };
 
+  const startSentMessageProcess = async () => {
+    await handleSentQueryRequest();
+  };
+
```

Comment:
I was also thinking on this part too, as why should the footer have any responsibility over this? My idea is to just call one method in the client and that's it. All of the useQuery set states to the client as well. If that makes sense?

Reply:
Everyting from handleSentQueryRequest() is now moved to QueryCopilotClient.